### PR TITLE
apply clang-tidy fixes

### PR DESCRIPTION
**Pull request by tjhei**

### CODE REVIEW

File: source/adiabatic_conditions/function.cc [hunk 1]
```diff
@@ -29,12 +29,12 @@ namespace aspect
...
     Function<dim>::Function()
       : function (3)
-    {}
+        = default;
 
     template <int dim>
     void
     Function<dim>::initialize()
-    {
+  {
```
Comment (gassmoeller) [inline source/adiabatic_conditions/function.cc:37]: this looks wrong

File: source/geometry_model/sphere.cc [hunk 2]
```diff
@@ -33,15 +33,15 @@ namespace aspect
...
+        = default;
 
 
 
     template <int dim>
     void
     Sphere<dim>::
     create_coarse_mesh (parallel::distributed::Triangulation<dim> &coarse_grid) const
-    {
+  {
```
Comment (gassmoeller) [inline source/geometry_model/sphere.cc:44]: same here

File: source/geometry_model/spherical_shell.cc [hunk 3]
```diff
@@ -54,13 +54,13 @@ namespace aspect
...
       spherical_manifold()
-    {}
+        = default;
 
     template <int dim>
     void
     SphericalShell<dim>::
     create_coarse_mesh (parallel::distributed::Triangulation<dim> &coarse_grid) const
-    {
+  {
```
Comment (gassmoeller) [inline source/geometry_model/spherical_shell.cc:63]: and here

File: source/initial_temperature/lithosphere_mask.cc [hunk 4]
```diff
@@ -35,14 +35,14 @@ namespace aspect
...
-      {}
+          = default;
 
 
 
       template <int dim>
       void
       LABDepthLookup<dim>::initialize ()
-      {
+    {
```
Comment (gassmoeller) [inline source/initial_temperature/lithosphere_mask.cc:45]: and here

File: source/material_model/rheology/constant_viscosity.cc [hunk 5]
```diff
@@ -34,13 +34,13 @@ namespace aspect
...
         viscosity(numbers::signaling_nan<double>())
-      {}
+          = default;
 
 
 
       double
       ConstantViscosity::compute_viscosity () const
-      {
+    {
```
Comment (gassmoeller) [inline source/material_model/rheology/constant_viscosity.cc:43]: indent

File: source/simulator/stokes_matrix_free.cc [hunk 6]
```diff
@@ -668,12 +668,12 @@ namespace aspect
...
     :
     MatrixFreeOperators::Base<dim, dealii::LinearAlgebra::distributed::BlockVector<number> >()
-  {}
+      = default;
 
   template <int dim, int degree_v, typename number>
   void
   MatrixFreeStokesOperators::StokesOperator<dim,degree_v,number>::clear ()
-  {
+{
```
Comment (gassmoeller) [inline source/simulator/stokes_matrix_free.cc:676]: and here

**Review by gassmoeller — COMMENTED** [review]
Looks good, except for the few places with funny indentation. Please fix, afterwards this is good to go.

**Comment by tjhei** [comment]
Huh, the auto fixes are a complete mess and don't even compile...

**Comment by tjhei** [comment]
I removed all the nonsensical stuff for now...

**Review by gassmoeller — APPROVED** [review]

File: source/postprocess/boundary_pressures.cc [hunk 7]
```diff
@@ -110,10 +110,10 @@ namespace aspect
         const char *columns[] = { "Pressure at top (Pa)",
                                   "Pressure at bottom (Pa)"
                                 };
-        for (unsigned int i=0; i<sizeof(columns)/sizeof(columns[0]); ++i)
+        for (auto &column : columns)
```
Comment (bangerth) [inline source/postprocess/boundary_pressures.cc:113]: Oh, I didn't even know this works. How nice!

**Review by bangerth — APPROVED** [review]

**Comment by bangerth** [comment]
Can clang-tidy convert old-style loops into range-based for loops? If so, I'd accept a patch to that effect!

**Comment by tjhei** [comment]
There is https://clang.llvm.org/extra/clang-tidy/checks/modernize-loop-convert.html and it should be enabled, but I don't see anything reported...

**Pull request merged**